bridge/src/observers/nine-chronicles.ts

```diff
@@ -61,8 +61,35 @@ export class NCGTransferredEventObserver implements IObserver<{ blockHash: Block
                 const minimum = new Decimal(this._limitationPolicy.minimum);
                 const maximum = new Decimal(this._limitationPolicy.maximum);
                 const transferredAmountInLast24Hours = new Decimal(await this._exchangeHistoryStore.transferredAmountInLast24Hours("nineChronicles", sender));
+                const maxExchangableAmount = maximum.sub(transferredAmountInLast24Hours);
+                const limitedAmount: Decimal = Decimal.min(maxExchangableAmount, amount);
 
-                if (recipient === null || !isValidAddress(recipient) || !amount.isFinite() || amount.isNaN()) {
+                const lessThanMinimum = amount.cmp(minimum) === -1;
+                const alreadyExchangedUptoMaximum = transferredAmountInLast24Hours.cmp(maximum) >= 0;
+                const overflowedExchangeAmount = !alreadyExchangedUptoMaximum && transferredAmountInLast24Hours.add(amount).cmp(maximum) === 1;
+
+                const isInvalidRecipient = recipient === null || !isValidAddress(recipient);
+                const isInvalidAmount = !amount.isFinite() || amount.isNaN();
+                const isInvalidTx = isInvalidRecipient || isInvalidAmount;
+
+                const isExchangableTx = !isInvalidTx && !lessThanMinimum && !alreadyExchangedUptoMaximum
```

Comment:
Note: I pre-calculated the conditions to record exchange history with amount before execution.